libyul/optimiser/ASTWalker.h

```diff
@@ -130,7 +129,7 @@ struct ForEach: Base
 template<typename Node, typename Entry, typename Visitor>
 void forEach(Entry&& _entry, Visitor&& _visitor)
 {
-	detail::ForEach<Node, std::decay_t<Visitor>>{std::forward<Visitor>(_visitor)}(std::forward<Entry>(_entry));
+	detail::ForEach<Node, Visitor&>{_visitor}(_entry);
```

Comment:
Actually, we also don't need to forward ``_entry`` - we know that it will always be passed either by reference or const reference anyways.